File: backend/src/main/java/com/ric/backend/model/Event.java
package com.ric.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Data
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "venue_id")
    private Venue venue;

    private String title;
    private String description;
    private String eventType;
    private String bannerImage;
    private LocalDateTime eventDate;
    private LocalTime startTime;
    private LocalTime endTime;
    private String status;
    private BigDecimal ticketPrice;
    private LocalDateTime createdAt;
}

File: frontend/src/types/event.ts
/**
 * Event and Venue types
 * Covers: Events listing, Venue information
 */

/**
 * Venue information
 */
export interface Venue {
  id: number;
  name: string;
  totalCapacity: number;
  area: string;
  venueType: string;
}

/**
 * Event details
 * Status can be: "LIVE", "UPCOMING", "PAST", "CANCELLED"
 */
export interface Event {
  id: number;
  title: string;
  description: string;
  eventType: string;
  bannerImage: string;
  eventDate: string;
  startTime: string;
  endTime: string;
  status: "LIVE" | "UPCOMING" | "PAST" | "CANCELLED";
  ticketPrice: number;
  venue: Venue;
}

/**
 * Event list response (array of events)
 */
export type EventListResponse = Event[];
